classes/services/exceptions/InvalidSubmissionException.php

```diff
@@ -0,0 +1,42 @@
+<?php 
+
+/**
+ * @file classes/services/exceptions/InvalidSubmissionException.php
+ *
+ * Copyright (c) 2014-2017 Simon Fraser University
+ * Copyright (c) 2000-2017 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class InvalidSubmissionException
+ * @ingroup services_exceptions
+ *
+ * @brief Invalid submission exception class
+ */
+
+namespace App\Services\Exceptions;
+
+class InvalidSubmissionException extends ServiceException {
+	
+	/** @var int Submission ID */
+	protected $submissionId = null;
+	
+	/**
+	 * Constructor
+	 *
+	 * @param string $message
+	 * @param int $code
+	 */
+	public function __construct ($contextId, $submissionId) {
+		$this->submissionId = $submissionId;
+		parent::__construct($contextId, "Invalid submission");
```

Comment:
I think this class also starts to impinge on the authorization framework -- see e.g. https://github.com/pkp/pkp-lib/blob/master/classes/security/authorization/internal/SubmissionRequiredPolicy.inc.php. We might enhance the auth framework to use exceptions (we've avoided those so far), for example, but I worry about diluting that toolset with similar things.